Dockerfile

```diff
@@ -32,23 +37,17 @@ VOLUME /data
 RUN apt-get update
 RUN DEBIAN_FRONTEND=noninteractive apt-get upgrade -y
 
+# We want to use Ubuntu's stock rsyslog rather than syslog-ng
+# that the base image provides.
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y rsyslog
+RUN rm -rf /etc/service/syslog-ng
+
 # Install packages needed by Mail-in-a-Box.
 ADD containers/docker/apt_package_list.txt /tmp/mailinabox_apt_package_list.txt
 RUN DEBIAN_FRONTEND=noninteractive apt-get install -y $(cat /tmp/mailinabox_apt_package_list.txt)
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y rsyslog
-RUN rm -f /tmp/mailinabox_apt_package_list.txt
-RUN apt-get clean
-
-# Create the user-data user, so the start script doesn't have to.
-RUN useradd -m user-data
 
 # Now add Mail-in-a-Box to the system.
 ADD . /usr/local/mailinabox
 
-# We can't know things like the IP address where the container will eventually
-# be deployed until the container is started. We also don't want to create any
-# private keys during the creation of the image --- that should wait until the
-# container is started too. So our whole setup process is deferred until the
-# container is started.
-RUN mkdir -p /etc/my_init.d
-RUN ln -s /usr/local/mailinabox/containers/docker/init.sh /etc/my_init.d/20-mailinabox.sh
+# Cleanup
+RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

Comment:
This doesn't do what you intend. It just adds another image layer which contains these deletions. So the resulting image size is even bigger than without this command. 

Apart from that, the base image is derived from official Ubuntu image. That one is configured to automatically clean up after every `apt-get install`. So running `apt-get clean && rm -rf /var/lib/apt/lists/*` manually should not be needed at all.